.github/workflows/ci.ml

```diff
@@ -296,13 +301,19 @@ let main_build_job ~analyse_job ~cygwin_job ?section runner start_version ~oc ~w
   (* Intentionally fail fast, no need to run all build if there is a
    * problem in a given version; usually it is functions not defined in lower
    * versions of OCaml. *)
-  let (_fail_fast, matrix, _) = platform_ocaml_matrix ~fail_fast:true start_version in
   let (matrix, includes) =
     if platform = Windows then
-      (("host", ["x86_64-pc-cygwin"; "i686-w64-mingw32"; "x86_64-w64-mingw32"; "i686-pc-windows"; "x86_64-pc-windows"]) ::
-       ("build", ["x86_64-pc-cygwin"]) ::
-       matrix, [])
+      ([], [
+          [("host", "x86_64-pc-cygwin"); ("build", "x86_64-pc-cygwin"); ("ocamlv", latest_ocaml4)];
+          [("host", "i686-w64-mingw32"); ("build", "x86_64-pc-cygwin"); ("ocamlv", latest_ocaml4)];
+          [("host", "x86_64-w64-mingw32"); ("build", "x86_64-pc-cygwin"); ("ocamlv", latest_ocaml4)];
+          [("host", "x86_64-w64-mingw32"); ("build", "x86_64-pc-cygwin"); ("ocamlv", latest_ocaml5)];
+          [("host", "i686-pc-windows"); ("build", "x86_64-pc-cygwin"); ("ocamlv", latest_ocaml4)];
+          [("host", "x86_64-pc-windows"); ("build", "x86_64-pc-cygwin"); ("ocamlv", latest_ocaml4)];
+          (* [("host", "x86_64-pc-windows"); ("build", "x86_64-pc-cygwin"); ("ocamlv", latest_ocaml5)]; 5.3 needed *)
+        ])
```

Comment:
What about something like that, it would be easier to read the essential information : host its ocaml version, especially if its supposed to grow in the future:
```suggestion
      let matrix =
        let ocaml4 = [
          "x86_64-pc-cygwin";
          "i686-w64-mingw32";
          "x86_64-w64-mingw32";
          "i686-pc-windows";
          "x86_64-pc-windows"
        ] in
        let ocaml5 = [
          "x86_64-w64-mingw32";
          (* "x86_64-pc-windows"; 5.3 needed *)
        ] in
        let matrix_elem ocamlv hosts =
          let elem ocaml host =
            [("host", host); ("build", "x86_64-pc-cygwin"); ("ocamlv", ocaml)]
          in
          List.map (elem ocamlv) hosts
        in
        matrix_elem latest_ocaml4 ocaml4
        @ matrix_elem latest_ocaml5 ocaml5
      in
      ([], matrix)
```
or
```suggestion
      let matrix =
        let hosts = [
          (* host,            ocaml4, ocaml5 *)
          "x86_64-pc-cygwin",   true, false;
          "i686-w64-mingw32",   true, false;
          "x86_64-w64-mingw32", true, true;
          "i686-pc-windows",    true, false;
          "x86_64-pc-windows",  true, false;  (* 5.3 needed *)
        ] in
        let matrix_elem ocamlv host =
          [("host", host); ("build", "x86_64-pc-cygwin"); ("ocamlv", ocamlv)]
        in
        List.map (fun (host, ocaml4, ocaml5) ->
            (if ocaml4 then matrix_elem latest_ocaml4 host else [])
            :: (if ocaml5 then matrix_elem latest_ocaml5 host else [])
            :: [])
          hosts
        |> List.flatten
        |> List.filter (function [] -> false | _ -> true)
      in
      ([], matrix)